### ISSUE TITLE
Issue with local caching - Add a warning about it?

### ISSUE THREAD

**Issue by Vadorequest** [mxstbr/react-boilerplate#1074]
# Issue with local caching

## Issue Type

- [ ] Feature: Add a warning/message about local caching (FAQ?) due to "offline-first" feature.

## Description

I ran into something very weird today. Basically, I didn't get the same output when hitting F5 and ctrl+F5. F5 showed the old website, and ctrl+F5 showed me the latest -correct- version. I couldn't figure out why, local caching was disabled through the browser console. It's only when @gihrig gave me this solution that I finally figured out (or think so) what was wrong:

`Chrome dev tools > Application > Clear Storage > Clear site data`

I think that because of the "offline-first" feature, *sometimes*, the local caching gets screwed. And the only way to fix it is to clear it.

Basically, a simple warning in the docs somewhere would be enough, I don't really know what's the best place for it (maybe when we talk about offline-first? Assuming it's indeed related to offline-first.).

## Steps to reproduce

Couldn't really reproduce this error, happened once, hasn't happened again since.

# Versions

- Node/NPM: 6.7.0/3.10.3
- Browser:  53.0.2785.143 m (64-bit)

**Comment by gihrig** [comment]
To reproduce (just a guess) :
In new clone with Example app code still in working dir:
`npm run start:production`
Open in Chrome browser.
Stop server
_now example app and its service worker are loaded in browser_

`npm run clean`
`npm start`
Open in Chrome browser
_Loading from server will get cleaned app. Loading from offline-first will get example app._

Also see #852 especially this [comment](https://github.com/mxstbr/react-boilerplate/issues/852#issuecomment-239920419).

**Comment by Vadorequest** [comment]
Yep, this seems to be exactly what I have done! :)

I guess we should simply add it to the react-boilerplate FAQ, because people will likely run into this. If it's related to the production mode, maybe that's where it should be stated.

**Comment by gihrig** [comment]
@Vadorequest see [#1075 comment](https://github.com/mxstbr/react-boilerplate/issues/852#issuecomment-251825900)

**Comment by gihrig** [comment]
Discussion moved to #852 - Closing this.

**Comment by gihrig** [comment]
See updated [FAQ](https://github.com/mxstbr/react-boilerplate/blob/dev/docs/general/faq.md#issue-with-local-caching-when-running-in-production-mode-f5--ctrlf5--cmdr-weird-behavior)

**Comment by GGAlanSmithee** [comment]
A couple of questions, if you don't mind :)

Is this only a problem when running locally, or would you get the same problem when deploying under the same circumstances? Also, is the cache-related problem only present when updating during the same browser session?

Thanks!

**Comment by gihrig** [comment]
> Is this only a problem when running locally, or would you get the same problem when deploying under the same circumstances?

In my development only experience the problem is that Chrome loads the cached data without refreshing from the 'server'. This is the point of offline-first, but Chrome seems to favor cached data too heavily. I have not experienced these problems on Firefox or Safari on Mac.

Given that the problem is browser specific, I see no reason for it to be different in a production environment.

> Also, is the cache-related problem only present when updating during the same browser session?

According to the [document](https://github.com/NekR/offline-plugin/blob/master/docs/updates.md#making-sure-serviceworker-updates-at-all) linked in the [FAQ](https://github.com/mxstbr/react-boilerplate/blob/dev/docs/general/faq.md#issue-with-local-caching-when-running-in-production-mode-f5--ctrlf5--cmdr-weird-behavior)  the browser tab must be closed and a new tab opened to load the new data. See those documents for details.

My experience, in a development environment only, leads me to believe there may be a production problem with Chrome browsers, but I have not scrutinized the [configuration](https://github.com/NekR/offline-plugin/blob/master/docs/updates.md#config) details as above.

If you follow those details exactly and experience a problem please report it in a new issue.

**Comment by GGAlanSmithee** [comment]
Thanks for the clarfications @gihrig this makes it much clearer as to what's going on.

If I experience any unexpected behaviors, I'll comment on it, othervise thanks for your help.
